File: DailyChallenges/CPP/LeetCode_938.cpp
// 938. Range Sum of BST @ 2022/12/07

class Solution {
public:
    int rangeSumBST(TreeNode* root, int low, int high) {
        if (root){
            if (root->val < low)
                return rangeSumBST(root->right, low, high);
            else if (root->val > high)
                return rangeSumBST(root->left, low, high);
            else
                return root->val + rangeSumBST(root->right, low, high) + rangeSumBST(root->left, low, high);
        }else
            return 0;
    }
};

File: DailyChallenges/Scala/LeetCode_938.scala
object Solution {
    def rangeSumBST(root: TreeNode, low: Int, high: Int): Int = {
        if (root == null) 0 else {
            if (root.value > high)
               rangeSumBST(root.left, low, high)
            else if (root.value < low)
                rangeSumBST(root.right, low, high)
            else
                root.value + rangeSumBST(root.left, low, high) + rangeSumBST(root.right, low, high)
        }
    }
}
